### DISCUSSION TITLE
Formatting numbers in u8g2

### DISCUSSION THREAD

**Discussion by wjgray3** [olikraus/u8g2#2118] [Q&A] [answered]
Is it possible to format a floating point number in u8g2?  The u8g2.print command doesn't have any options for printing in any particular format, like say, a floating point number to only one digit after the decimal.  It appears to always print two after the decimal point.  
After much checking, I don't see a simple way.
Any help?

**Comment by olikraus** [comment, marked as answer]
U8g2.print is just the same as arduino.print.
Use sprintf to do proper formating of floats.

**Reply by wjgray3** [reply]
Can you give me an example please?
u8g2.sprintf doesn't appear to be supported.  You must have meant something else.

Suppose I want a 16 bit float "SOG" to be displayed on my lcd with one digit to the right of the decimal.

I've tried u8g2.print(sprint(%SOG,1));

That didn't work.

Still trying other options.

**Reply by olikraus** [reply]
sprintf is a generic float to string conversion function. It is described here: https://www.tutorialspoint.com/c_standard_library/c_function_sprintf.htm
Idea is to convert the float into a sequence of chars, representing the desired number, it usually looks like this:
```
  float f = 1.234;
  char buf[10];
  sprintf(s, "%.2f", f);  // .2 means: print 2 numbers after the dot
```

Once done, just print "buf" with u8g2.print(buf);

**Reply by wjgray3** [reply]
Okay, that worked great, except that in your example, the "s" in the sprintf  function above needs to be "buf", the name of the buffer.
End result, it works great.

I didn't realize that to print a formatted number, I needed to change it to a string of characters in the format I want, and then print the string.

Thank you very very much.
Bill

Now if I can only get the display contrast working properly.........

**Reply by olikraus** [reply]
Ah, yes... s must be buf

**Reply by olikraus** [reply]
I will look into the contrast topic later
